### DISCUSSION TITLE
Github Action using Docker image doesn't have plugins in the Docker Image

### DISCUSSION THREAD

**Discussion by MichaelPStimpson** [community/community#49828] [Actions] [answered]
### Question

Why aren't my installed plugins showing up inside a Docker container while running on a Github Runner?

### Body

When I use a docker file on my local machine I see the expected plugins, but when I use it with a Github Action it does NOT show the plugins. I suspect it might be a path issue with the plugins, but it's difficult to debug inside a Github Runner. (and it's unclear why the path changes between my local machine and the Github Runner.)

Any ideas? 
Additional information below!

The CLI tool is SFDX and the plugins are: sfdx-git-delta, sfdx-scanner, and sfdx-hardis.
Here is the YML file:
```
jobs:
  test-script:
    runs-on: ubuntu-latest
    container: michaelpstimpson/sf-pipeline:base
    steps:
      - name: Check for plugins
        run: |
          sfdx plugins #returns "no plugins installed."
```

The Dockerfile itself is quite simple:

```
FROM salesforce/salesforcedx:latest-full

RUN sfdx plugins install @salesforce/sfdx-scanner
RUN echo 'y' | sfdx plugins install sfdx-hardis
RUN echo 'y' | sfdx plugins install sfdx-git-delta
```

The parent dockerfile:

```
FROM heroku/heroku:20

ENV DEBIAN_FRONTEND=noninteractive
ARG SALESFORCE_CLI_VERSION=nightly
ARG SF_CLI_VERSION=latest-rc

RUN echo 'b298a73a9fc07badfa9e4a2e86ed48824fc9201327cdc43e3f3f58b273c535e7  ./nodejs.tar.gz' > node-file-lock.sha \
  && curl -s -o nodejs.tar.gz https://nodejs.org/dist/v18.15.0/node-v18.15.0-linux-x64.tar.gz \
  && shasum --check node-file-lock.sha
RUN mkdir /usr/local/lib/nodejs \
  && tar xf nodejs.tar.gz -C /usr/local/lib/nodejs/ --strip-components 1 \
  && rm nodejs.tar.gz node-file-lock.sha

ENV PATH=/usr/local/lib/nodejs/bin:$PATH
RUN npm install --global sfdx-cli@${SALESFORCE_CLI_VERSION} --ignore-scripts
RUN npm install --global @salesforce/cli@${SF_CLI_VERSION}

RUN apt-get update && apt-get install --assume-yes openjdk-11-jdk-headless jq
RUN apt-get autoremove --assume-yes \
  && apt-get clean --assume-yes \
  && rm -rf /var/lib/apt/lists/*

ENV SFDX_CONTAINER_MODE true
ENV DEBIAN_FRONTEND=dialog
ENV SHELL /bin/bash
```

heroku/heroku:20 can be found [here](https://hub.docker.com/layers/heroku/heroku/20-build.v30/images/sha256-424378b498b74adfe9d3e046fc9e135bc7a7195839639efaa6d7e8104a454bd3?context=explore)

**Comment by MichaelPStimpson** [comment, marked as answer]
I was able to find the plugin installation directories on my local machine inside the docker container via `sfdx plugins inspect *plugin-name*`. I suspect it may be because SFDX and the plugins were installed on my local machine before spinning up the container. Regardless, calling `sfdx plugins link plugin-name` from inside the plugin base directory linked the plugins with the container in the Github runner and resolved the issue!

**Comment by oberegovskyi** [comment]
@MichaelPStimpson  hey, can you please add additional details, of how you were able to solve it? I having exactly the same problem, I want to extend sfdx docker with a couple of installed plugins, however, plugins are not available/visible in GitHub actions.
Much thanks for those details

**Comment by alcabon** [comment]
The other workaround is to set the env HOME variable with /root.

 - name: Plugins are available
        run: |
            export HOME=/root
            echo $HOME
            echo "Plugins list START:"
            sfdx plugins
            echo "Plugins list END"
